### PR TITLE
Detail function of each file in component documentation

### PR DESCRIPTION
**Pull request by romaricpascal**
As I was trying to understand what each file was responsible for in the components, I thought I'd log it in the documentation so it's easier for the next person starting to work on the project. Let me know if I missunderstood or missed something 😊

### CODE REVIEW

File: docs/contributing/coding-standards/components.md [hunk 1]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
-      - `macro.njk`
-      - `template.njk`
-      - `template.test.js`
+- `README.md` - Light documentation pointing to installation instructions and component documentation on <https://design-system.service.gov.uk/>
+- `_[component-name].scss` - A SCSS file to generate the styles just for the component, importing the variables, functions and mixins from `../../base` and the styles of the component (`./index.scss`).
+- `_index.scss` - The actual styles for the component, that can be imported either on their own by `[component-name].scss` or alongside other components in `components/_all.scss` (or by the consuming application)
+- `[component-name].yaml` - A list of the options expected by the component, used for listing in the documentation 
+- `macro.njk` - The main entry point for rendering the component, provides a `govuk[ComponentName](params)` macro, delegating render to the `template.njk` file
+- `template.njk` - The template used for rendering the component using any `params` provided to the macro
+- `template.test.js` - Tests to ensure the component renders as intended with its various option
```
Comment (owenatgov) [inline docs/contributing/coding-standards/components.md:19]:
```suggestion
- `template.test.js` - Tests to ensure the component renders as intended with its various options
```

**Review by owenatgov — COMMENTED** [review]
I've made one super tiny nitpick but this looks great all the same. Thanks for writing it up!

I'm gonna approve this in word only for now though as I think it's worth @claireashworth running their tech writer eye over it to make sure the tech content's in good shape.

File: docs/contributing/coding-standards/components.md [hunk 2]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
 
-      - `README.md`
-      - `_[component-name].scss`
-      - `_index.scss`
-      - `[component-name].yaml`
-      - `macro.njk`
-      - `template.njk`
-      - `template.test.js`
+- `README.md` - Light documentation pointing to installation instructions and component documentation on <https://design-system.service.gov.uk/>
+- `_[component-name].scss` - A SCSS file to generate the styles just for the component, importing the variables, functions and mixins from `../../base` and the styles of the component (`./index.scss`).
```
Comment (domoscargin) [inline docs/contributing/coding-standards/components.md:14]:
```suggestion
- `_[component-name].scss` - An SCSS file to generate the styles just for the component, importing the variables, functions and mixins from `../../base` and the styles of the component (`./index.scss`).
```

File: docs/contributing/coding-standards/components.md [hunk 3]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
+- `_index.scss` - The actual styles for the component, that can be imported either on their own by `[component-name].scss` or alongside other components in `components/_all.scss` (or by the consuming application)
+- `[component-name].yaml` - A list of the options expected by the component, used for listing in the documentation
```
Comment (domoscargin) [inline docs/contributing/coding-standards/components.md:16]: This file also includes examples of the component with specific options. These are used to generate examples in the documentation, as well as to run tests on.

Comment (romaricpascal) [reply docs/contributing/coding-standards/components.md]: Thanks for clarifying that, I hadn't noticed it. Is that file also listing the sets of options used for the fixtures that Ollie was mentionning this morning?

Comment (domoscargin) [reply docs/contributing/coding-standards/components.md]: Yup, those are used to generate the fixtures during certain Gulp tasks: https://github.com/alphagov/govuk-frontend/blob/54272096c11bf73c6e1fa5dd992e01c5e42a8a93/tasks/gulp/copy-to-destination.js#L56

Comment (romaricpascal) [reply docs/contributing/coding-standards/components.md]: Cheers, I'll add that to the descrption as well, then.

File: docs/contributing/coding-standards/components.md [hunk 4]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
-      - `_[component-name].scss`
-      - `_index.scss`
-      - `[component-name].yaml`
-      - `macro.njk`
-      - `template.njk`
-      - `template.test.js`
+- `README.md` - Light documentation pointing to installation instructions and component documentation on <https://design-system.service.gov.uk/>
+- `_[component-name].scss` - An SCSS file to generate the styles just for the component, importing the variables, functions and mixins from `../../base` and the styles of the component (`./index.scss`).
+- `_index.scss` - The actual styles for the component, that can be imported either on their own by `[component-name].scss` or alongside other components in `components/_all.scss` (or by the consuming application)
+- `[component-name].yaml` - Lists the options expected by the component, used for display in the documentation. It also the option samples used to generate the examples in the documentation, as well as the [fixtures for testing alternative implementations of the design system](https://frontend.design-system.service.gov.uk/testing-your-html/).
```
Comment (domoscargin) [inline docs/contributing/coding-standards/components.md:16]:
```suggestion
- `[component-name].yaml` - Lists the options expected by the component, used for display in the documentation. It also contains the option samples used to generate the examples in the documentation, as well as the [fixtures for testing alternative implementations of the design system](https://frontend.design-system.service.gov.uk/testing-your-html/).
```

File: docs/contributing/coding-standards/components.md [hunk 5]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
 When creating your component, you should create the following files in the component’s folder:
 
-      - `README.md`
-      - `_[component-name].scss`
-      - `_index.scss`
-      - `[component-name].yaml`
-      - `macro.njk`
-      - `template.njk`
-      - `template.test.js`
+- `README.md` - Light documentation pointing to installation instructions and component documentation on <https://design-system.service.gov.uk/>
```
Comment (claireashworth) [inline docs/contributing/coding-standards/components.md:13]:
```suggestion
- `README.md` - Concise documentation with links to the installation instructions and component documentation on <https://design-system.service.gov.uk/>
```

File: docs/contributing/coding-standards/components.md [hunk 6]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
+- `_[component-name].scss` - An SCSS file to generate the styles just for the component, importing the variables, functions and mixins from `../../base` and the styles of the component (`./index.scss`).
```
Comment (claireashworth) [inline docs/contributing/coding-standards/components.md:14]:
```suggestion
- `_[component-name].scss` - An SCSS file to generate the styles for the component, import the variables, functions and mixins from `../../base` and the styles of the component (`./index.scss`).
```

File: docs/contributing/coding-standards/components.md [hunk 7]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
+- `_index.scss` - The actual styles for the component, that can be imported either on their own by `[component-name].scss` or alongside other components in `components/_all.scss` (or by the consuming application)
```
Comment (claireashworth) [inline docs/contributing/coding-standards/components.md:15]:
```suggestion
- `_index.scss` - The actual styles for the component that you can import in 2 ways:

-  on their own using `[component-name].scss`
- alongside other components in `components/_all.scss` (or with the consuming application)

```

File: docs/contributing/coding-standards/components.md [hunk 8]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
+- `[component-name].yaml` - Lists the options expected by the component, used for display in the documentation. It also contains the option samples used to generate the examples in the documentation, as well as the [fixtures for testing alternative implementations of the design system](https://frontend.design-system.service.gov.uk/testing-your-html/).
```
Comment (claireashworth) [inline docs/contributing/coding-standards/components.md:16]:
```suggestion
- `[component-name].yaml` - Lists the component's Nunjucks macro options, which are added to the component's documentation. It also contains the option samples used to generate the examples in the documentation, as well as the [fixtures for testing alternative implementations of the design system](https://frontend.design-system.service.gov.uk/testing-your-html/).
```

File: docs/contributing/coding-standards/components.md [hunk 9]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
+- `macro.njk` - The main entry point for rendering the component, provides a `govuk[ComponentName](params)` macro, delegating render to the `template.njk` file
```
Comment (claireashworth) [inline docs/contributing/coding-standards/components.md:17]:
```suggestion
- `macro.njk` - The main entry point for rendering the component. It provides a `govuk[ComponentName](params)` macro, delegating render to the `template.njk` file
```

File: docs/contributing/coding-standards/components.md [hunk 10]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
+- `template.njk` - The template used for rendering the component using any `params` provided to the macro
+- `template.test.js` - Tests to ensure the component renders as intended with its various options
 
 If your component uses JavaScript, you must also create the following files in the component’s folder:
 
-      - `[component-name].mjs`
-      - `[component-name].unit.test.mjs`
-      - `[component-name].test.js`
+- `[component-name].mjs` - A JavaScript module with the implementation of any behaviour needed by the component. See the [JavaScript documentation]('./js.md#skeleton) for a skeleton and more details on that file's structure
+- `[component-name].unit.test.mjs` - Unit tests to verify any lower-level logic used by the component individually
```
Comment (claireashworth) [inline docs/contributing/coding-standards/components.md:24]: 'individually' is an orphan in this sentence. Is it still accurate to say 'Does individual unit tests to verify...' ?

File: docs/contributing/coding-standards/components.md (hunk 8 above)
Comment (domoscargin) [reply docs/contributing/coding-standards/components.md:16]: While it's correct that these are the Nunjucks macro options, we could possibly be more generic here, for the theoretical case where we ditch Nunjucks in favour of something else, these options would probably still be needed for whatever else we decided to use.

Maybe something like "Lists the parameters that can be passed to the component" (or less of a mouthful!).

File: docs/contributing/coding-standards/components.md (hunk 10 above)
Comment (domoscargin) [reply docs/contributing/coding-standards/components.md:24]: Could maybe reword as "Unit tests to verify any component-specific lower-level logic"?

Basically, these are testing behaviour used internally by this component only.

File: docs/contributing/coding-standards/components.md (hunk 6 above)
Comment (romaricpascal) [reply docs/contributing/coding-standards/components.md:14]: I think this edit loses the meaning I was trying to express here, to help distinguish this file from `_index.scss`. This file is used so that people using the `govuk-frontend` package can [import the styles of a single component on its own](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-an-individual-component-s-css-using-a-single-import). 

Re-reading it, it feels like it's not really relevant how it does it (the part about importing the variables,...) as a developer will see that pretty clearly when they open the file, and there may be other parts getting added later on that'd require us to update that doc.  Maybe just that it delegates the actual styling to the `_index.scss` file to give people a heads up that there's not much going on in that file.

Maybe something like:
> An SCSS file to generate the styles for this component only. It delegates the CSS generation to the `_index.scss` file (see right after).

File: docs/contributing/coding-standards/components.md (hunk 10 above)
Comment (romaricpascal) [reply docs/contributing/coding-standards/components.md:24]: @domoscargin That sounds much better than the way I phrased it initially 😄

File: docs/contributing/coding-standards/components.md (hunk 6 above)
Comment (claireashworth) [reply docs/contributing/coding-standards/components.md:14]: Ah yes, that is very clear. I don't think the bracket info at the end is needed but I'll amemd the suggested edit to your text.

File: docs/contributing/coding-standards/components.md (hunk 8 above)
Comment (claireashworth) [reply docs/contributing/coding-standards/components.md:16]: OK, sounds sensible. I like your new sentence, but do you think it would be worth adding the Nunjucks macro as an example. Maybe like - "Lists the parameters (for example, Nunjucks macros) that are passed to the component." ? This sentence format also assumes that they definitely are passed - is that right or are they a possible, rather than a definite, action?

File: docs/contributing/coding-standards/components.md (hunk 10 above)
Comment (claireashworth) [reply docs/contributing/coding-standards/components.md:24]: Yup, that's fab.

File: docs/contributing/coding-standards/components.md (hunk 8 above)
Comment (claireashworth) [reply docs/contributing/coding-standards/components.md:16]: Yes, that's the one. I'll edit the suggestion to your description.

File: docs/contributing/coding-standards/components.md [hunk 11]
```diff
@@ -10,19 +10,19 @@ Generally, folder and file names should be singular, for example ‘accordion’
...
```
Comment (claireashworth) [inline docs/contributing/coding-standards/components.md:24]:
```suggestion
- `[component-name].unit.test.mjs` - Unit tests to verify any component-specific lower-level logic.
```

File: docs/contributing/coding-standards/components.md (hunk 8 above)
Comment (domoscargin) [reply docs/contributing/coding-standards/components.md:16]: I've tweaked my wording - I was conflating the Design System website (which does use the options in its documentation), with this repo specifically.

File: docs/contributing/coding-standards/components.md (hunk 7 above)
Comment (romaricpascal) [reply docs/contributing/coding-standards/components.md:15]:
```suggestion
- `_index.scss` - The actual styles for the component that you can import in 2 ways -  [on their own by `[component-name].scss`](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-specific-parts-of-the-css) or [alongside other components in `components/_all.scss`](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-specific-parts-of-the-css)
```

**Comment by domoscargin** [comment]
I'll let @claireashworth approve the final change, but this looks good to go from me. Nice!

File: docs/contributing/coding-standards/components.md (hunk 7 above)
Comment (claireashworth) [reply docs/contributing/coding-standards/components.md:15]: @romaricpascal this looks good, thank you.

**Review by claireashworth — APPROVED** [review]
